kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/arbitrary/DateTest.kt

```diff
@@ -212,4 +214,46 @@ class DateTest : WordSpec({
          checkAll(10) { _: Period -> /* No use. Won't reach here if unsupported */ }
       }
    }
+
+   "Arb.yearMonth(minYearMonth, maxYe)" should {
+      "generate valid YearMonths (no exceptions)" {
+         Arb.yearMonth().take(10_000).toList()
+      }
+
+      "generate YearMonths between minYear and maxYear" {
+         val years = mutableSetOf<Int>()
+         val months = mutableSetOf<Int>()
+
+         checkAll(10_000, Arb.yearMonth(YearMonth.of(1998, 1), YearMonth.of(1999, 12))) {
+            years += it.year
+            months += it.monthValue
+         }
+
+         years shouldBe setOf(1998, 1999)
+         months shouldBe (1..12).toSet()
```

Comment:
I think this test needs improvement.

We're not guaranteeing here that we're not generating YearMonths from outside the range specified